test/integration/targets/copy/tasks/tests.yml

```
@@ -2448,3 +2448,44 @@
       loop:
         - '{{ src }}'
         - '{{ src }}_dest'
+
+- name: Record the output directory
+  set_fact:
+    remote_file: foo.txt
```

We could keep these files contained to the remote_tmp_dir.

```suggestion
    remote_file: "{{ remote_tmp_dir }}/foo.txt"
    ansible_remote_tmp: "{{ remote_tmp_dir }}"
```